### ISSUE TITLE
如何使用/mail指令发送物品？| How can I send items using the /mail command?

### ISSUE THREAD

**Issue by QSlotus** [Melledy/LunarCore#76]
如题 | Same topic.

**Comment by yuko1101** [comment]
You can attach items with `<itemId>:<amount>`.
Try this command, `/mail 102:100 hello`

**Comment by QSlotus** [comment]
谢谢，那么我该如何使用/mail指令获取自定义圣遗物呢？
比如说，我需要使用/mail获取一个自定义圣遗物。如果用/give来获取它指令是 `/give 61054 lv15 s4 4:1 5:2 8:2 9:4`
***
Thanks a lot, so how do I get a custom holy relic using the /mail command?
Let's say I need to get a custom holy relic using /mail. If I use /give to get it the command is `/give 61054 lv15 s4 4:1 5:2 8:2 9:4`

**Comment by QSlotus** [comment]
Also, how do I change the title of a message sent using /mail so that it is not `Test`?
![image](https://github.com/Melledy/LunarCore/assets/109905194/9b9d0916-be07-49ab-852c-00b7074e0f35)

**Comment by yuko1101** [comment]
The title is hardcoded in MailCommand, so you cannot change the title with /mail.
And as for relic as attachment seems to be partly impossible since mail attachments cannot have substats info atleast.

**Comment by yuko1101** [comment]
I tried to attach this relic (can be reproduce by running `/give @10001 61043 s5 2:1 4:1 5:1 7:6 lv15 -maxsteps`)
![image](https://github.com/Melledy/LunarCore/assets/68993883/885e266c-5783-4925-8a83-a67bca284f56)

The mail I received
![image](https://github.com/Melledy/LunarCore/assets/68993883/b8e910f0-70d9-480d-aae3-c4a0d3963128)

Attached item
![image](https://github.com/Melledy/LunarCore/assets/68993883/0c5e751e-08b1-477e-96ad-09fa79a1c485)

correct mainstat but not for substats

**Comment by yuko1101** [comment]
oh my bad, its also corrent for substats

**Comment by yuko1101** [comment]
TL;DR: You can send custom relic with custom plugins but not with /mail.

You can't see the substats before you take attachments since client does not have substats data, but process of sending items is handled on the server, so it have nothing to do with proto.

**Comment by yuko1101** [comment]
I just created a plugin for this. https://github.com/yuko1101/CustomMailPlugin
Put the plugin jar into `plugins` folder and restart server.
Then run `/cmail A mail with a relic | You | Here is a relic for you! $(61043 s5 2:1 4:1 5:1 7:6 lv15 -maxsteps)`.

**Comment by QSlotus** [comment]
Thank you! The problem has been solved. By the way, how can I use the `serverOptions.defaultPermissions` field in `config.json` to restrict player permissions? For example, I only want them to use the `/help`, `/give`, `/giveall` commands and nothing else!

**Comment by yuko1101** [comment]
just remove "*" and put in "player.give" and "player.help"

**Comment by QSlotus** [comment]
Thank you! It worked properly.
Also, do you have any ideas about this issue ? (#74 )

**Comment by QSlotus** [comment]
Also, I'd like to ask, when will the skill tree feature and [swarm disaster] be fixed for the simulated universe?
